### DISCUSSION TITLE
accessing formData outside of dependsOn?

### DISCUSSION THREAD

**Discussion by jaxramus** [laravel/nova-issues#4403] [Q&A] [answered]
I have a CRM I am working on where a `customer` can create their own additional `fields`.

What I've done to make this happen is I have a table called `fields` with a `type` column, and then a panel iterates through the `customer` `fields` in order to show the appropriate Nova field based on their `type`.

My problem is I have a `leads` resource which contains a `customer` `BelongsTo` relation, and I want the `lead` create/update form to show the `fields` dynamically when the `customer` BelongsTo value changes.

Currently it appears the only way I can get nova to update a form is by using `dependsOn`, but what I really need is access to something like `formData` for the whole resource itself, and I need this resource-level `formData` to change when the `customer` `BelongsTo` field changes.

Is this possible via any type of workaround?

**Comment by crynobone** [comment, marked as answer]
`FormData` is only available within `dependsOn`.
